feat: Live Activities Validation & Events Display Enhancement

mohdashraf010897 opened the pull request:
# Overview

https://jira.corp.adobe.com/browse/MOB-24006
This PR adds **Live Activities validation** and **event display** for Assurance Live Activities plugin.
---

## Key Features
- **Validation:** Client info card updates, iOS/push detection, clear platform messaging, real-time status  
- **Activities Display:** List of registered Live Activities, event timelines, status tracking (active, completed, dismissed)  
- **Event Management:** Resizable details panel, copy support, improved navigation, filtering & search  

---

## Technical Improvements
- Restructured plugin state for clearer navigation  
- Centralized reusable components in `@assurance/common-utils`  
- UX updates: subtle CTAs, spacing, intuitive flow  
- Stronger type safety with TypeScript types & constants

dsoffiantini committed:
feat: :sparkles: innovation week changes

dsoffiantini committed:
live activities

sagar-sharma-adobe committed:
Adding state selectors and event filters for Validation page

nsamra-adobe-dev committed:
add: app config validation

sagar-sharma-adobe committed:
Adding client filter component and changes.

sagar-sharma-adobe committed:
Changing implementation for ECID and Client Selection

nsamra-adobe-dev committed:
add: app credential validation

nsamra-adobe-dev committed:
feat: data stream validation

sagar-sharma-adobe committed:
Creating Client Validation logic and widget

sagar-sharma-adobe committed:
Refactoring and simplification

sagar-sharma-adobe committed:
Client Picker effect fix. Code restructuring

sagar-sharma-adobe committed:
Adding property data api from launch-toolkit for property data. Added property checks to credentials

mohdashraf010897 committed:
MOB-23940: Assurance LA & PN - Validation Page : Adding live activity validation to client info card

- Added Live Activities validation section to client validation widget
- Created reusable utility functions for iOS version validation logic
- Implemented conditional table rendering based on support status
- Added tooltips for each table row with contextual information
- Extracted business logic to pure utility functions for better testability
- Added comprehensive unit tests for validation utilities
- Moved constants to constants.tsx for better organization
- Refactored hooks to focus on React concerns while delegating business logic to utils

mohdashraf010897 committed:
refactor: Apply best practices to Live Activities validation code

- Extract hard-coded values to constants file (COPYABLE_VALUE_CONSTANTS, VALIDATION_STATUS)
- Create separate clipboard utility functions in utils/clipboard.ts
- Extract CopyableValue to reusable component with proper TypeScript interfaces
- Replace all hard-coded strings with localized messages using react-intl
- Make CopyableValue component accept localized tooltip messages as props
- Remove duplicate code (getHealthIcon) and organize imports properly
- Localize table row labels (iOS Version, Device Type, etc.)
- Extract validation status constants to reusable constants
- Fix import errors and TypeScript type issues
- Improve code organization with proper separation of concerns

All code now follows React, TypeScript, and localization best practices.

mohdashraf010897 committed:
fix: Complete localization of all hard-coded strings in Live Activities validation

- Add missing localized messages for all remaining hard-coded strings:
  - 'Unknown' for fallback values
  - 'Yes'/'No' for boolean values
  - 'None detected'/'registered' for activity counts
  - 'Not available' for missing tokens
  - Tooltip messages for activity status
- Update createLiveActivitiesTableData to accept localized 'Not available' text
- Update all test files to use VALIDATION_STATUS constants
- Ensure 100% localization coverage with no hard-coded user-facing strings

All user-facing text is now properly localized using react-intl.

mohdashraf010897 committed:
fix: Resolve all test failures and improve code quality

- Fix validation utility edge cases in parseIOSVersion and isDeviceVersionBelowAppMinimum
- Update App component tests to use proper React Testing Library queries
- Update graphql-request from 7.1.0 to 7.2.0 to fix dependency issues
- Remove empty test file causing 'No test suite found' error
- Fix duplicate 'unknown' property in localization messages
- All 22 tests now passing with 100% success rate

Test Results:
✅ Live Activities Display Tests: 6/6 passed
✅ Live Activities Validation Tests: 11/11 passed
✅ Live Activities App Tests: 4/4 passed
✅ Content Cards Tests: 1/1 passed

mohdashraf010897 committed:
fix: improve error handling in CopyableValue component

- Add proper try-catch block for clipboard operations
- Handle clipboard errors gracefully without crashing the app
- Add console.warn for debugging clipboard failures
- Ensure copied state is only set on successful copy operations

mohdashraf010897 committed:
feat: add constants and test IDs for Live Activities validation

- Add testIds.ts with comprehensive test identifiers for all components
- Add constants for copy functionality and validation status
- Organize constants for better maintainability and testing
- Support for both registered and active activities testing

mohdashraf010897 committed:
test: add comprehensive CopyableValue component tests

- Add 16 test cases covering all functionality
- Test basic functionality: rendering, truncation, null handling
- Test copy functionality: clipboard operations and visual feedback
- Test customization: maxLength, showFullValue, custom tooltips
- Test accessibility: ARIA labels and test IDs
- Test visual states: copy icon, checkmark, timeout behavior
- Test error handling: graceful clipboard error handling
- All tests passing with proper error handling

mohdashraf010897 committed:
feat: implement proper empty state handling using renderEmptyState

- Replace manual empty state rows with Adobe React Spectrum renderEmptyState prop
- Add dedicated empty state renderer functions for registered and active activities
- Simplify table rendering logic by removing conditional empty state rendering
- Follow Adobe React Spectrum best practices for TableView empty states
- Improve maintainability and consistency with design system
- Support both basic and full support scenarios with appropriate empty states

mohdashraf010897 committed:
test: add comprehensive Live Activities validation section tests

- Add 7 test cases covering all validation scenarios
- Test basic support: section headings and status display
- Test full support: full support status and section rendering
- Test not supported: proper handling of unsupported devices
- Test error handling: graceful handling of missing/null data
- Test accessibility: proper ARIA labels and test IDs
- All tests passing with proper mock data setup
- Focus on testable functionality given Adobe React Spectrum limitations

mohdashraf010897 committed:
test: add utility function tests for Live Activities

- Add tests for liveActivitiesSchemaExtraction utility
- Add tests for liveActivitiesStateExtraction utility
- Ensure proper test coverage for data extraction functions
- Support testing of core Live Activities data processing logic

mohdashraf010897 committed:
feat: enhance Live Activities core functionality and data handling

- Update App.tsx with improved Live Activities integration
- Enhance useActivities hook with better data management
- Update liveActivities types for improved type safety
- Improve liveActivitiesExtraction utility functions
- Remove deprecated useLiveActivitiesData hook
- Consolidate data handling for better maintainability

mohdashraf010897 committed:
refactor: remove deprecated useLiveActivitiesData hook

- Remove useLiveActivitiesData.ts as functionality consolidated into useActivities
- Clean up deprecated code to improve maintainability
- Consolidate Live Activities data handling into single hook

mohdashraf010897 committed:
fix: improve registered activities detection for iOS 16.4 using Live Activity Schema events

- Add extractRegisteredActivitiesFromSchemaEvents function to use schema events as source of truth
- Update extractLiveActivitiesDataFromState to prioritize schema events over shared state
- Enhance useLiveActivitiesData to always include schema events for better detection
- Add comprehensive tests for schema-based extraction functionality
- Fix issue where registered activities weren't detected for iOS 16.4
- Schema events now serve as primary source of truth for registered activities
- Fallback to shared state only when no schema events are available
- All 82 tests passing with improved registered activities detection

mohdashraf010897 committed:
fix: remove trailing whitespace in useActivities.ts

mohdashraf010897 committed:
feat: add Live Activity Schema event matchers for comprehensive event capture

- Add 'Live Activity Schema' event name pattern matcher using regex (~)
- Include schema event matchers in ALL_LIVE_ACTIVITIES for complete coverage
- Update SCHEMA_EVENTS matcher to include event name pattern
- Remove debug console.log from useLiveActivitiesData hook
- Ensures events like 'Live Activity Schema (AirplaneTrackingAttributes)' are captured
- All tests passing with improved event detection coverage

mohdashraf010897 committed:
fix: remove unsupported regex matcher for Rambda compatibility

- Remove '=~' regex operator that doesn't work with Rambda matcher
- Keep jsonSchema and examplePayload matchers which are sufficient
- These matchers will catch Live Activity Schema events by their data structure
- Fixes LexerError: Unknown character: ~ in event filtering
- All tests passing with working matchers

mohdashraf010897 committed:
Remove Active Live Activities section from validation table

- Removed Active Live Activities table section and related UI components
- Cleaned up unused message definitions and empty state handlers
- Removed useActivities hook dependency from validation component
- Updated tests to remove Active Live Activities assertions
- Maintained Registered Live Activities section functionality

mohdashraf010897 committed:
feat: enhance Live Activities UI with new components and improved functionality

- Add new ActivityCard, ActivityDetails, and ActivityList components for better activity management
- Add SpectrumCard atom component for consistent card styling
- Add activities-enhanced container for improved activity handling
- Update useActivities hook with enhanced activity management capabilities
- Improve live activity components (activity-events, activity-info, activity-timeline, launch-live-activity)
- Update liveActivities types and extraction utilities
- Enhance sample app components with improved Live Activities integration
- Update event data viewer and event table sample apps

mohdashraf010897 committed:
refactor: improve pushToStart token extraction and clean up unused code

- Add useSelectedClientPushToStartToken hook following same pattern as push token
- Move pushToStart token from registered activities table to main status table
- Remove pushToStart token column from registered activities table for better UX
- Remove unused extractPushToStartToken utility function and Rambda dependency
- Remove unused LIVE_ACTIVITIES_FIELD_LABELS constant
- Add localized tooltip messages for pushToStart token
- Use direct client state extraction: payload.metadata['state.data'].liveActivity.pushToStartToken
- Maintain consistency with existing client data extraction patterns

mohdashraf010897 committed:
refactor: replace custom copy implementations with CopyableValue component

- Remove custom CopyToClipboard implementations for pushToStartToken and updateToken
- Replace with consistent CopyableValue component for better UX and maintainability
- Remove unused imports: ActionButton, CopyIcon, CopyToClipboard
- All copyable values now use the same component with consistent tooltips and truncation

mohdashraf010897 committed:
feat: add comprehensive Live Activity end event support

- Add LiveActivityEndedEvent interface for 'Live Activity ended' events
- Create isLiveActivityEndedEvent type guard function
- Add isLiveActivityEndEvent combined type guard for both dismissed and ended events
- Update useActivities hook to detect both dismissed and ended events
- Update event matchers to include 'Live Activity ended' events
- Fix activity status logic to properly show 'completed' for end events
- Add console logging for debugging end event detection
- Update event type helper function to recognize 'ended' events

This ensures activities show as 'completed' when either 'Live Activity dismissed'
or 'Live Activity ended' events are present, providing more robust end state detection.

mohdashraf010897 committed:
feat: add PushToStart token to profile section with CopyableValue

- Add LiveActivityPushNotificationDetail interface with complete structure
- Update ProfileEntity to include liveActivityPushNotificationDetails
- Add PushToStart Token row to profile section with CopyableValue component
- Show PushToStart Token only for iOS devices with full support (≥17.1)
- Add proper TypeScript types for attributeType and denylisted fields
- Include localized copy functionality with tooltips and feedback
- Maintain consistent UI patterns with existing profile fields

mohdashraf010897 committed:
feat(@assurance/event-data-viewer): Add tooltip support for overflowing values

- Add showTooltips and tooltipThreshold props to DataViewer and DataViewerTable
- Implement tooltip rendering for values exceeding character threshold
- Add TooltipTrigger and Tooltip components from React Spectrum
- Clean up debug console logs
- Fix syntax error in DataViewerCell component

This enhancement improves UX by showing full content of long values on hover,
making the data viewer more user-friendly for complex event payloads.

mohdashraf010897 committed:
feat(@assurance/event-table): Implement resizable event details panel with advanced UX

Major Features:
- Add EventTableWithDetails component with slide-in resizable panel
- Implement ResizeHandle with dynamic grip positioning
- Add EventDetailsPanel with React Spectrum Accordion for organized content
- Integrate @assurance/event-data-viewer for enhanced payload display
- Add CopyableValue and CopyableMonacoEditor components for data copying

Technical Improvements:
- Create useGripPosition hook for intelligent resize handle positioning
- Add useResizePanel hook for panel width management with constraints
- Implement ResizeObserver for container height change detection
- Add useDocumentStyles hook for resize state styling
- Support accordion collapse/expand with automatic grip repositioning

UX Enhancements:
- Dynamic grip positioning within middle 60% of resize bar
- Visual feedback with color changes during resize
- Smooth transitions and hover effects
- Copy functionality with visual feedback (checkmark icon)
- Tooltip support for overflowing values in event data
- Responsive design with proper constraints

Dependencies:
- Update @adobe/react-spectrum to ^3.38.0 for Accordion support
- Add @monaco-editor/react for JSON editing
- Integrate @assurance/event-data-viewer for data visualization

This implementation provides a professional, resizable event details view
similar to @bootstrap-event-list/ with enhanced UX and maintainability.

mohdashraf010897 committed:
chore: Update dependencies for event-table and event-data-viewer

- Update yarn.lock with new React Spectrum 3.38.0 dependency
- Add package-lock.json for npm dependency management
- Resolve peer dependency conflicts with --legacy-peer-deps

These dependency updates support the new Accordion component
and ensure compatibility across the event-table package ecosystem.

mohdashraf010897 committed:
feat: enhance Live Activities UX with modular architecture and resizable event details

- Add resizable event details panel with slide-in functionality
- Implement centralized constants and configuration management
- Create master hook (useLiveActivitiesMaster) for consolidated data access
- Add event processing utilities for filtering, statistics, and time ranges
- Introduce shared type definitions to resolve TypeScript conflicts
- Create reusable components: MetricCard, InfoField, ContentStateCard
- Add enhanced clipboard functionality with copy feedback
- Implement ErrorBoundary for better error handling
- Update CopyableValue to use new clipboard hook
- Refactor ActivityEventDetails to use new utilities and centralized messages
- Clean up debug logs and improve code organization
- Fix TypeScript type conflicts between LiveActivityEvent definitions

Components:
- @assurance/event-table: Enhanced with resizable details panel
- @assurance/event-data-viewer: Added tooltip support for overflowing values
- @live-activities: Complete modular architecture refactor

This commit significantly improves code maintainability, reusability, and user experience.

mohdashraf010897 committed:
refactor: clean up old Activities tab and rename Activities Enhanced to Activities

- Remove old Activities tab from App.tsx navigation
- Rename Activities Enhanced container to Activities
- Delete redundant components: ActivityInfo, ActivityTimeline, ActivityEvents
- Remove unused SCSS files: activities.scss, activity-info.scss
- Update all imports and references to use new Activities container
- Clean up internationalization messages and remove old references
- Simplify tab structure to: Client Info, Activities, Events

This consolidates the Live Activities functionality into a single, enhanced tab
with better UX and removes duplicate/redundant code.

mohdashraf010897 committed:
feat: improve Live Activities UX with better card layout and meaningful status filters

- Update ActivityCard layout to prioritize Live Activity ID over attribute type
  - Live Activity ID now displayed prominently with monospace font
  - Attribute type shown as secondary info in smaller, italic text
  - Better visual hierarchy for unique identification

- Remove meaningless 'Inactive' filter from activity list
  - Live Activities only have 'Active' and 'Completed' statuses
  - Updated type definitions to remove 'inactive' status
  - Cleaned up status handling logic across all components
  - Removed unused message definitions

- Improve status logic consistency
  - Updated LiveActivity interface to only include meaningful statuses
  - Simplified status variant handling in ActivityCard and ActivityDetails
  - Updated constants to reflect actual Live Activity lifecycle

This makes the interface more intuitive by highlighting the unique identifier
and removing confusing filter options that don't apply to Live Activities.

mohdashraf010897 committed:
feat: improve resize panel functionality and UI enhancements

- Fix resize direction issue in EventTableWithDetails.tsx
  - Add panelPosition parameter to useResizePanel hook
  - Support both 'left' and 'right' panel positions
  - Right panels now resize correctly (drag right = decrease width)

- Consolidate resize panel hook
  - Move useResizePanel from event-table to common-utils
  - Remove duplicate hook to prevent conflicts
  - Export TypeScript interfaces for better type safety

- Enhance activities UI
  - Improve 'No activities found' placeholder size and visibility
  - Increase font sizes, padding, and spacing for better UX
  - Replace HTML elements with React Spectrum components
  - Remove debug information from production UI

- Add new utility hooks to common-utils
  - useResizeObserver for container width tracking
  - useDocumentStyles for resize state management
  - Centralized hook exports for better organization

- Update TypeScript configurations
  - Fix storybook tsconfig for better type resolution
  - Ensure proper type exports across packages

mohdashraf010897 committed:
fix: resolve event count discrepancy and React hooks violation

- Fix event count discrepancy between ActivityCard and EventDetails
- Add deduplication logic to useActivityEvents and extractActiveActivitiesFromEvents
- Update extractActivityMetadata to use flexible matching logic for activityId
- Add 'Live Activity update token to Edge' event type support
- Remove 'push token to Edge' event type (not activity-specific)
- Fix React hooks rule violation in ActivityOverview component
- Remove all debugging console logs
- Ensure consistent event counting across all components

Resolves event count mismatch where ActivityCard showed fewer events than EventDetails.
All components now show the same deduplicated event count.

mohdashraf010897 committed:
feat: centralize CopyableValue component and enhance copy functionality

- Move CopyableValue to common-utils package for reusability across all packages
- Add comprehensive props support (copyTooltip, copyFullValueTooltip, copiedMessage)
- Remove duplicate CopyableValue components from event-table and live-activities packages
- Add React Spectrum dependencies to common-utils package
- Enhance CopyableValue with label support, truncation, and multiple button types
- Update all components to use centralized CopyableValue
- Simplify EventDataViewer to show only payload copy button
- Remove redundant 'Raw Event JSON' label from EventDetailsPanel
- Maintain backward compatibility with existing functionality

mohdashraf010897 committed:
feat: restructure plugin state and improve ContentStateCard CTA

- Restructure usePluginState with hierarchical navigation organization
- Add NAVIGATION_CONFIG constants for better type safety and maintainability
- Separate top-level navigation (main tabs) from activity-level navigation
- Improve ContentStateCard CTA design with subtle ViewDetail button
- Add scroll-to-event functionality in EventTableWithDetails
- Remove scroll logic from EventTable for better separation of concerns
- Update all components to use new state structure with proper typing
- Export ActivityTab and TopLevelTab types for better developer experience

mohdashraf010897 committed:
refactor: reorganize imports in ContentStateCard component

sagar-sharma-adobe committed:
Adding experience url redirection to util

mohdashraf010897 committed:
Fix code review issues

- Separate React hooks from pure utility functions in eventProcessing.ts
- Fix deprecated document.execCommand with proper error handling
- Consolidate duplicated copy logic in ContentStateCard
- Replace string literals with EVENT_CONFIG constants in activity-flow.tsx
- Fix Math.random() keys with deterministic index-based keys
- Extract navigation logic from Zustand store into reusable utilities

All fixes maintain existing functionality while improving code organization,
testability, and maintainability.

mohdashraf010897 committed:
Update code review fixes

- Refine navigation utilities implementation
- Improve clipboard error handling
- Update event processing exports

mohdashraf010897 committed:
Fix formatting in useActivityEvents.ts

mohdashraf010897 committed:
fix: resolve TypeScript configuration issues for event-table storybook

- Fix tsconfig.storybook.json to include component files for story imports
- Remove project reference from main tsconfig.json to avoid noEmit conflict
- Add type casting for defaultColumns in stories to resolve type errors
- Ensure Storybook builds work correctly with proper TypeScript configuration

mohdashraf010897 committed:
fix: address all Copilot AI code review comments

- Move dynamic clipboard import to module level in ContentStateCard
- Improve error handling in DatastreamHorizontal for non-string messages
- Fix parseIOSVersion return type to handle NaN cases with null return
- Refactor getValidSize to use type guard instead of repeated type assertions
- Extract IIFE in activities.tsx to useMemo hook for better readability

All changes maintain existing functionality while improving code quality and type safety.

mohdashraf010897 committed:
fix: address additional Copilot AI code review comments

- Fix logical inconsistency in parseIOSVersion by removing redundant || 0
- Improve Monaco editor paste handling with debounced approach instead of hardcoded timeout
- Memoize JSON.stringify in ContentStateCard Monaco editor to avoid expensive recalculations

All changes improve performance and code quality while maintaining existing functionality.

mohdashraf010897 committed:
improve: enhance parseIOSVersion utility function

- Add comprehensive input validation for edge cases (null, empty, non-string)
- Trim whitespace from version strings before parsing
- Validate that major version is non-negative
- Properly handle missing minor version (defaults to 0)
- Validate minor version is non-negative when provided
- Add more comprehensive JSDoc examples
- Improve error handling and robustness

The function now handles all edge cases gracefully and provides better type safety.

mohdashraf010897 committed:
test: add comprehensive test suite for liveActivitiesValidation utilities

- Add 23 test cases covering parseIOSVersion function
- Test all edge cases: valid inputs, invalid inputs, special characters
- Test getLiveActivitiesSupport with different iOS versions
- Test validateIOSVersionForLiveActivities with various inputs
- Test isDeviceVersionBelowAppMinimum comparison logic
- All tests pass and provide 100% coverage of utility functions
- Ensures robust error handling and correct behavior validation

mohdashraf010897 committed:
fix: address additional Copilot AI code review comments and improve test coverage

- Fix Monaco editor type in launch-live-activity.tsx to use IStandaloneCodeEditor
- Improve module organization in liveActivitiesConfig.ts by keeping MESSAGES as standalone export
- Fix ref type assertion in activities.tsx by making useResizeObserver hook generic
- Fix setTimeout type issue in useResizeObserver hook (NodeJS.Timeout vs number)
- Add comprehensive test coverage for getStatusDisplayConfig function
- Enhance parseIOSVersion utility with better error handling and validation
- Add 24 comprehensive test cases covering all utility functions with 100% coverage

All changes improve type safety, code organization, and maintainability while ensuring robust test coverage.

mohdashraf010897 committed:
refactor: remove unnecessary CopyableValue re-export file

- Remove packages/ui-plugins/live-activities/src/components/atoms/CopyableValue.tsx
- Remove associated test file CopyableValue.test.tsx
- Update all imports to use @assurance/common-utils directly
- Eliminate code duplication and unnecessary abstraction layer

This addresses Copilot AI feedback about removing files that only contain re-exports with extensive comments and empty lines.

mohdashraf010897 committed:
Convert CSS modules from SCSS to regular CSS

- Renamed all .module.scss files to .css files
- Updated all import statements to use .css instead of .module.scss
- Simplified TypeScript declarations to only handle .css files
- Removed SCSS preprocessing dependency
- Maintained all existing styling functionality
- Improved build performance and simplicity

mohdashraf010897 committed:
Fix CSS import statements

- Updated all .module.css imports to .css
- Resolved build errors caused by incorrect import paths
- Build now works correctly with regular CSS files

mohdashraf010897 committed:
Update yarn.lock after CSS conversion

- Updated lockfile to reflect changes from SCSS to CSS conversion
- Removed SCSS-related dependencies
- Lockfile now matches the simplified CSS-only setup

mohdashraf010897 committed:
Add missing CSS files and fix TypeScript declarations

- Created ContentStateCard.css with all required styles
- Updated types.d.ts to include both .css and .module.css declarations
- Resolved TypeScript module resolution errors
- All CSS imports now work correctly

mohdashraf010897 committed:
Address Copilot AI code review comments

- Update parseIOSVersion documentation to clarify patch versions are ignored
- Remove unnecessary try-catch block in extractAttributeTypeFromSchemaEvent
- Optimize sortEventsByTimestamp to cache Date objects for better performance
- Fix type assertion in usePluginState navigateToEvent function
- Memoize lowercase search query in ActivityList filtering to avoid repeated conversions

All changes improve code quality, performance, and type safety.

mohdashraf010897 committed:
Improve CSS class handling with classnames package

- Add classnames import to all components using CSS classes
- Replace string literals with classNames() calls for better readability
- Maintain consistent class handling throughout the package
- Improve conditional class logic with classNames utility
- All components now use proper classNames() function calls

This provides better readability, maintainability, and consistency
when handling CSS classes, especially for conditional styling.

mohdashraf010897 committed:
Add missing classNames import to client-info.tsx

- Add classNames import to maintain consistency with other components
- Ensures proper class handling throughout the validation components

mohdashraf010897 committed:
Address Copilot AI code review comments

- Add missing classNames import to launch-live-activity.tsx
- Improve parseIOSVersion robustness with whitespace trimming
- Replace raw div with Spectrum View in activities.tsx for consistency
- Add ARIA labels and roles for better accessibility in validation section
- Fix ref type compatibility by using div for resize container
- Update aria-label to use statusConfig.title instead of label

All changes improve code quality, accessibility, and maintainability.

mohdashraf010897 committed:
fix: Add missing dependencies and build script for live-activities plugin

- Add react-dom, @adobe/react-spectrum, @react-aria/*, @spectrum-icons/workflow, react-intl, uuid dependencies
- Fix build script to use Parcel with production optimizations
- Remove main field to avoid Parcel target conflicts
- Enable proper production builds for OSS plugin integration

nsamra-adobe-dev committed:
feat: redirection buttons in validations

nsamra-adobe-dev committed:
Merge branch 'feat/live-activity-experience-redirection' into ns-la-redirection-changes

mohdashraf010897 committed:
fix: resolve classNames import and build issues in live-activities plugin

- Add missing classNames import to App.tsx to fix ReferenceError
- Fix TypeScript type error in onSelectionChange callback
- Add proper build script with Parcel configuration
- Add missing dependencies (react-dom, @adobe/react-spectrum, etc.)
- Add TypeScript checking scripts for better development experience
- Remove invalid main field from package.json
- Add tsconfig.json for proper TypeScript validation

This resolves the runtime error 'classNames is not defined' that occurred
in the minified production bundle.

nsamra-adobe-dev committed:
fix: refactor

nsamra-adobe-dev committed:
fix: unit tests

nsamra-adobe-dev committed:
Merge branch 'feat/MOB-23940/live-activities-client-validation' into ns-la-redirection-refactor

nsamra-adobe-dev committed:
add: husky precommit hook

nsamra-adobe-dev committed:
Merge pull request #10 from adobe/ns-la-redirection-refactor

Ns la redirection refactor

mohdashraf010897 committed:
fix: add token fields to LiveActivityDisplayRow and fix test failures

- Add pushToStartToken and updateToken fields to LiveActivityDisplayRow type
- Update createLiveActivitiesTableData to populate token fields based on validation status
  - Basic support (iOS 16.1-16.3): pushToStartToken shows "Not available"
  - Full support (iOS 16.4+): shows actual token values or "Not available"
- Remove examplePayload from schema extraction to match expected test structure
- All 219 tests now passing

mohdashraf010897 commented:
Trimmed down version concerning LA changes only - https://github.com/adobe/assurance-plugins/pull/11

cc @dsoffiantini @nsamra-adobe-dev 

When you guys have confidence in the trimmed-down PR, then we can close this PR

nsamra-adobe-dev committed:
fix: comment send/update LA

dsoffiantini commented on package-lock.json:
This can probably be deleted for yarn

dsoffiantini commented on lib/setup.js:
```diff
@@ -0,0 +1,24 @@
+import "@testing-library/jest-dom";
```
it looks like this is just a duplicate of the test setup?

dsoffiantini commented on packages/common/common-utils/src/clipboard.util.ts:
````diff
@@ -0,0 +1,121 @@
+ * };
+ * ```
+ */
+export async function copyToClipboard(text: string): Promise<boolean> {
````
Usually we just use react-copy-to-clipboard which can simplify this.

dsoffiantini commented on packages/ui-plugins/validation-summary/utils/groupBy.ts:
```diff
@@ -0,0 +1 @@
+export const groupBy = () => {};
```
no-op can be deleted

dsoffiantini commented on packages/ui-plugins/validation-summary/package.json:
this can probably be removed as im not sure if its complete

dsoffiantini commented on packages/ui-plugins/validation-editor/package.json:
This can be removed for now

dsoffiantini commented on packages/ui-plugins/optimize/package.json:
This one is also a wip and can be removed

mohdashraf010897 committed:
fix: address code review feedback from dsoffiantini

- Remove package-lock.json (yarn project doesn't need it)
- Remove lib/setup.js (duplicate of test/setup.ts)
- Remove unused clipboard.util.ts from common-utils
- Remove incomplete/WIP plugins: validation-summary, validation-editor, optimize

All changes address review comments from PR #XXX

mohdashraf010897 committed:
Merge pull request #12 from adobe/ns-send-update-comment-changes

fix: comment send/update LA

nsamra-adobe-dev committed:
fix: update CopyableValue component and refine useResizeObserver hook

jkartoshka commented:
I ran an assurance session with iOS version `18.4`, but with status` Live Activities not supported`. When I go to `Activities` tab I am told I need to update iOS version to be above 16.1 even though I already have 18.4. Can you update this to reflect the actual issue?
<img width="474" height="904" alt="Screenshot 2025-10-15 at 1 34 56 PM" src="https://github.com/user-attachments/assets/cd6a525d-8972-401b-a742-5c811dbf35eb" />
<img width="1427" height="712" alt="Screenshot 2025-10-15 at 11 57 09 AM" src="https://github.com/user-attachments/assets/5b55e5b4-7fc0-4d9e-bfa3-212524abb79d" />

jkartoshka commented on vite.config.ts:
```diff
@@ -0,0 +1,20 @@
+import react from "@vitejs/plugin-react-swc";
```
Since testing has been migrated to vite, I think `jest.config.js` & `babel-jest.config.js` can be removed.

nsamra-adobe-dev committed:
fix: activity tab conditional rendering

nsamra-adobe-dev committed:
fix: conditional rendering and UTs

nsamra-adobe-dev committed:
fix: client change handling

nsamra-adobe-dev committed:
Merge pull request #13 from adobe/ns-activity-tab-rendering-handle

fix: activity tab conditional rendering

nsamra-adobe-dev committed:
feat: remote start la

nsamra-adobe-dev committed:
fix: la start update payload, toast

nsamra-adobe-dev committed:
remove: console.logs statements

nsamra-adobe-dev committed:
fix: formatting and ts fixes

nsamra-adobe-dev committed:
fix: common folder

nsamra-adobe-dev committed:
add: i18n changes

nsamra-adobe-dev committed:
fix: more i18n fixes

nsamra-adobe-dev commented:
> I ran an assurance session with iOS version `18.4`, but with status` Live Activities not supported`. When I go to `Activities` tab I am told I need to update iOS version to be above 16.1 even though I already have 18.4. Can you update this to reflect the actual issue? <img alt="Screenshot 2025-10-15 at 1 34 56 PM" width="474" height="904" src="https://private-user-images.githubusercontent.com/51965286/501709864-cd6a525d-8972-401b-a742-5c811dbf35eb.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NjI3NjU1NzgsIm5iZiI6MTc2Mjc2NTI3OCwicGF0aCI6Ii81MTk2NTI4Ni81MDE3MDk4NjQtY2Q2YTUyNWQtODk3Mi00MDFiLWE3NDItNWM4MTFkYmYzNWViLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTExMTAlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUxMTEwVDA5MDExOFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWFjMzFkY2M1Y2Y0OTg4ZTkwOTVkZGZiNDQzOWNjMTUwNTk2ZDVhYjI3ZWVhNzY1OGMzNzc2OTdiMTVjZmEyNGQmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.aO4z2PQRiQS8nUD-1ezFrhUiCcYdW2TItQfcByKO85c"> <img alt="Screenshot 2025-10-15 at 11 57 09 AM" width="1427" height="712" src="https://private-user-images.githubusercontent.com/51965286/501709954-5b55e5b4-7fc0-4d9e-bfa3-212524abb79d.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NjI3NjU1NzgsIm5iZiI6MTc2Mjc2NTI3OCwicGF0aCI6Ii81MTk2NTI4Ni81MDE3MDk5NTQtNWI1NWU1YjQtN2ZjMC00ZDllLWJmYTMtMjEyNTI0YWJiNzlkLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTExMTAlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUxMTEwVDA5MDExOFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTJiYzU3Y2U3N2IwOWRmMTBhM2Y0MWEwYzFjMGUxMTVjOTA4ODliMDliMTE4MDA5MzhmZTY5YWQ4OGRlNzEyM2ImWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.6cDU3oUPBTfkRQDhC6DouEu6vf5E_vmi-TJDjSeFdDA">

Hey @jkartoshka, I think your app might be missing the build flags required for iOS Live Activities to run. You can find more details here: https://developer.apple.com/documentation/bundleresources/information-property-list/nssupportsliveactivities .
Also, I’ll update the status as "Full Live Activities Support" (for ios>18.1) and "Basic Live Activities Support" and messaging text in Activities tab, as LA flags should be taken care of on the app side to ensure LA support rather than in the SDK.

nsamra-adobe-dev committed:
fix: typo for activity id

nsamra-adobe-dev committed:
add: UTs for launch and update LA file

nsamra-adobe-dev committed:
Merge branch 'ns-send-update-changes' into ns-i18n-new-with-enhancements

nsamra-adobe-dev committed:
fix: css, activities &  label fixes

nsamra-adobe-dev committed:
fix: requested changes and ut fix

nsamra-adobe-dev committed:
Merge branch 'ns-send-update-changes' into ns-i18n-new-with-enhancements

nsamra-adobe-dev committed:
fix: query fixes

nsamra-adobe-dev committed:
remove: useMessages hook

nsamra-adobe-dev committed:
Merge pull request #15 from adobe/ns-send-update-refactor-with-i18n-changes

add: i18n changes

nsamra-adobe-dev committed:
fix: useEffect dependency

nsamra-adobe-dev committed:
Merge branch 'ns-send-update-changes' into ns-i18n-new-with-enhancements

nsamra-adobe-dev committed:
fix: review changes

nsamra-adobe-dev committed:
Merge pull request #17 from adobe/ns-i18n-new-with-enhancements

Ns i18n new with enhancements

nsamra-adobe-dev committed:
Merge pull request #14 from adobe/ns-send-update-changes

LA start, update and changes

nsamra-adobe-dev committed:
fix: realtime updates casing fix

nsamra-adobe-dev committed:
fix: resizer rerendering fix using useMemo

nsamra-adobe-dev committed:
fix: latest updated content state in editor

nsamra-adobe-dev committed:
fix: redundant key value

nsamra-adobe-dev committed:
fix: toast msg string fix

nsamra-adobe-dev committed:
fix: redundant code

nsamra-adobe-dev committed:
Merge pull request #18 from adobe/la-realtime-event-updates

La realtime event updates

nsamra-adobe-dev committed:
feat: display push-to-start tokens per activity type

nsamra-adobe-dev committed:
remove: commented code

nsamra-adobe-dev committed:
fix: ipad os version

nsamra-adobe-dev committed:
fix: error cases handling

nsamra-adobe-dev committed:
Merge pull request #20 from adobe/la-push-to-start-tokens-fix-ns

feat: display push-to-start tokens per activity type

nsamra-adobe-dev committed:
feat: Add initial broadcast live activity support

- Add broadcast type selection (unitary/broadcast) in launch modal
- Add broadcast channel ID input field for broadcast activities
- Update API to send broadcast type and channel ID in payloads
- Add type and broadcastChannelId fields to LiveActivity interface
- Update activity forms to handle broadcast vs unitary activities
- Disable type selection in update form (immutable after start)
- Add i18n messages for broadcast-related UI elements
- Extract type from start event data
- Broadcast activities don't require update token

Note: Known issue - activity.id conflates with channelID for broadcast
Will be fixed in next commit with proper separation

nsamra-adobe-dev committed:
fix: Resolve broadcast activity selection bug and add type filtering

BREAKING CHANGES:
- Activities now use unique keys (broadcast:channelID or unitary:activityID)
- This fixes the bug where all broadcast activities appeared selected
- Selection state is now properly maintained for both activity types

Features Added:
- Add getActivityKey() utility for unique activity identification
- Add visual type badges (🎮 Unitary / 📡 Broadcast) to activity cards
- Add optional type filter dropdown (All/Unitary/Broadcast)
- Type badges use color coding (blue for unitary, purple for broadcast)

Bug Fixes:
- Fix broadcast activities all showing as selected (empty id issue)
- Fix React key warnings for duplicate keys
- Fix activity selection not working for broadcast activities
- Ensure proper activity lookup by unique key

UX Improvements:
- Users can now filter by activity type
- Visual distinction makes it easy to identify activity types
- All activities shown by default for best discoverability

nsamra-adobe-dev committed:
fix: Address bugs introduced by broadcast support

Bug Fixes:
- Fix non-deterministic fallback key using Date.now()
  * Now uses stable 'no-timestamp' string instead
  * Prevents React re-render issues for edge case activities

- Fix user payload overwrite in buildCompleteApsPayload
  * Now merges user's custom fields with required broadcast fields
  * Preserves user data while adding channelID, origin, type

- Fix type detection to rely only on channelID presence
  * Removed non-existent eventData.type field check
  * Simplified logic: channelID present = broadcast, otherwise unitary

- Fix generateUpdateTemplate to only include existing fields
  * liveActivityID only added if activity.id exists
  * channelID only added if activity.broadcastChannelId exists
  * Prevents sending empty strings to backend

- Fix update modal heading to show channelID for broadcast
  * Now shows: id || broadcastChannelId || name
  * More user-friendly for broadcast activities

- Fix type filter counts to reflect search/status filters
  * Type counts now update based on visible activities
  * Prevents confusion when search reduces results

TODO Added:
- Added TODO comments for broadcast update token requirements
  * Need to verify if broadcast updates require tokens
  * Need to clarify what happens if pushToStartToken is missing

nsamra-adobe-dev committed:
Add type badge to activity cards and convert filters to dropdowns

nsamra-adobe-dev committed:
Minor UI and i18n updates for broadcast support

nsamra-adobe-dev committed:
Support multiple activity types per broadcast channel

- Update grouping key to use channelId:attributeType for broadcast
- Modify getActivityKey to include attributeType in composite key
- Enhance event filtering to match both channelID and attributeType
- Update activity card UI to show channel ID and activity type separately
- Fix event retrieval in overview and details components

nsamra-adobe-dev committed:
Refine ActivityCard UI and add missing type definitions

nsamra-adobe-dev committed:
Enable broadcast channel reuse for live activities

nsamra-adobe-dev committed:
Improve Start Activity button layout and UX

nsamra-adobe-dev committed:
fix: copilot review changes

nsamra-adobe-dev committed:
Add unit tests for broadcast live activity support

nsamra-adobe-dev committed:
Use constants for activity types and event types

nsamra-adobe-dev committed:
Fix missing imports and remove unused imports

nsamra-adobe-dev committed:
Add utility functions for activity key parsing

nsamra-adobe-dev committed:
request changes in review

nsamra-adobe-dev committed:
remove: console log

nsamra-adobe-dev committed:
refactor: extract splitActivityKeyParts utility

nsamra-adobe-dev committed:
Add broadcast UI test cases

nsamra-adobe-dev committed:
Merge pull request #21 from adobe/la-broadcast-support-refactor

La broadcast support refactor
